elixir/lib/mix/tasks/generate.ex

```diff
@@ -359,6 +379,53 @@ defmodule Mix.Tasks.Contracts.Generate do
     "object" not in types_list
   end
 
+  @doc """
+    Generates the cloud events conversion functions for the contract
+  """
+  def trento_cloudevent_conversion_functions() do
+    quote do
+      @doc """
+        Serialize and return a module struct from a cloud event json
+      """
+      def serialize_from_cloud_event(json_cloud_event) do
+        with {:ok, event} <- Cloudevents.from_json(json_cloud_event) do
+          create_contract_from_cloud_event(event)
+        end
+      end
+
+      @doc """
+        Serialize and return a cloud event json from a contract struct
+      """
+      def serialize_to_cloud_event(contract) do
+        with :ok <- validate_with_schema(contract),
+             {:ok, event} <-
+               CloudEvent.from_map(%{
+                 "specversion" => "1.0",
+                 "id" => UUID.uuid4(),
+                 "type" => @event_type,
+                 "source" => @source,
+                 "data" => contract
+               }) do
+          event
+        end
+      end
+
+      defp validate_with_schema(data) do
+        JsonSchema.validate(@event_type, data)
+      end
+
+      defp create_contract_from_cloud_event(%CloudEvent{data: data, type: @event_type}) do
+        with :ok <- validate_with_schema(data) do
+          __MODULE__.new(data)
```

Comment:
Does this already do the `base64` conversion?
Just curiosity

Reply:
Nope, after our discussion I think that it's better if we skip the encoding, I would change that in the golang generation too after we merge this PR :D